src/util/Permissions.js

```diff
@@ -32,12 +32,16 @@ class Permissions {
 
   /**
    * Gets all given permissions that are missing from the bitfield.
-   * @param {PermissionResolvable[]} permissions Permissions to check for
+   * @param {PermissionResolvable[]|PermissionResolvable} permissions Permissions to check for
    * @param {boolean} [checkAdmin=true] Whether to allow the administrator permission to override
-   * @returns {PermissionResolvable[]}
+   * @returns {Permissions}
    */
   missing(permissions, checkAdmin = true) {
-    return permissions.filter(p => !this.has(p, checkAdmin));
+    permissions = new this.constructor(~this.constructor.resolve(permissions) & this.bitfield);
```

Comment:
You're negating the wrong bitfield, it should be `this.constructor.resolve(permissions) & ~this.bitfield`